File: rxpermission/src/main/java/com/orca/kam/rxpermission/listener/PermissionListener.java
package com.orca.kam.rxpermission.listener;

import java.util.ArrayList;

/**
 * Project RxPermission
 *
 * @author Kang Young Won
 * @create 2016-08-29 - 오전 11:17
 */
public interface PermissionListener {

    /**
     * Call From Permission Activity...
     * When All Permissions Granted
     */
    void permissionGranted();

    /**
     * Call From Permission Activity....
     * When Permissions Denied
     *
     * @param deniedPermissions The Denied Permissions from PermissionActivity
     */
    void permissionDenied(ArrayList<String> deniedPermissions);
}

File: rxpermissionkt/src/main/java/com/orca/kam/rxpermissionkt/listener/PermissionListener.kt
package com.orca.kam.rxpermissionkt.listener

/**
 * Project RxPermission

 * @author Kang Young Won
 * *
 * @create 2016-08-29 - 오전 11:17
 */
interface PermissionListener {

    /**
     * Call From Permission Activity...
     * When All Permissions Granted
     */
    fun permissionGranted()

    /**
     * Call From Permission Activity....
     * When Permissions Denied

     * @param deniedPermissions The Denied Permissions from PermissionActivity
     */
    fun permissionDenied(deniedPermissions: List<String>)
}